### PR TITLE
refactor(storage): refactor l0 score calculate and decrease the priority of l0_intra

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

### <samp>🤖 Generated by Copilot at 3bcefba</samp>

Improved level 0 compaction logic in `TierCompactionPicker`. The compaction score and strategy now depend on the `LevelType` of each sub-level in level 0. This should enhance the read performance and workload distribution for level 0 compaction.

**This section will be used as the commit message. Please do not leave this empty!**

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)
- Refer to a related PR or issue link (optional)

-->

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/meta/src/hummock/compaction/level_selector.rs [hunk 1]
```diff
@@ -199,18 +198,24 @@ impl DynamicLevelSelectorCore {
...
+            // 3. base level comapction
+            // 4. l0 intra
+
             let l0_score =
                 idle_file_count as u64 * SCORE_BASE / self.config.level0_tier_compact_file_number;
             ctx.score_levels
                 .push((std::cmp::min(l0_score, max_l0_score), 0, 0));
-            let score = total_size * SCORE_BASE
-                / std::cmp::max(self.config.max_bytes_for_level_base, base_level_size);
-            ctx.score_levels.push((score, 0, ctx.base_level));
```
Comment (Little-Wallace) [inline src/meta/src/hummock/compaction/level_selector.rs:213]: Why change the score calculate

Comment (Li0k) [reply src/meta/src/hummock/compaction/level_selector.rs:213]: refactor it

File: src/meta/src/hummock/compaction/level_selector.rs [hunk 2]
```diff
@@ -193,24 +190,74 @@ impl DynamicLevelSelectorCore {
             .map(|level| level.table_infos.len())
             .sum::<usize>()
             - handlers[0].get_pending_file_count();
+
         let max_l0_score = std::cmp::max(
```
Comment (Little-Wallace) [inline src/meta/src/hummock/compaction/level_selector.rs:194]: do we need to remove `max_l0_score` ?

File: src/meta/src/hummock/compaction/level_selector.rs [hunk 3]
```diff
@@ -193,24 +190,74 @@ impl DynamicLevelSelectorCore {
...
+                    .iter()
+                    .filter(|level| level.level_type() == LevelType::Nonoverlapping)
+                    .count();
+
+                let non_overlapping_level_score = non_overlapping_level_count as u64 * SCORE_BASE
+                    / self.config.level0_sub_level_compact_level_count as u64;
+
+                std::cmp::min(
+                    std::cmp::max(non_overlapping_size_score, non_overlapping_level_score),
+                    max_l0_score,
```
[the remark this answers is not in the document]
Comment (Li0k) [reply src/meta/src/hummock/compaction/level_selector.rs:254]: can we remove the max_l0_score  whether a or b?

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Pull request merged**